src/main/java/net/minecraftforge/common/ForgeInternalHandler.java

```diff
@@ -135,5 +136,14 @@ public void resourceReloadListeners(AddReloadListenerEvent event)
         event.addListener(TierSortingRegistry.getReloadListener());
         event.addListener(CreativeModeTabRegistry.getReloadListener());
     }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void builtinMobSpawnBlocker(EntityJoinLevelEvent event)
+    {
+        if(event.getEntity() instanceof Mob mob && mob.isSpawnCancelled())
```

Comment:
This may be incorrect. `mob` is being assigned to the boolean value of the instance check. Then calling `isSpawnCancelled()`. I don't have the repo locally, nor any Java experience, but I do have 10 years of PHP experience. Got an error and this is where it lead me.

I'd imagine this should do the trick

```java
var mob = event.getEntity();
if (mob instanceof Mob && mob.isSpawnCancelled()) {
    event.setCanceled(true);
}
```

Reply:
```suggestion
    var mob = event.getEntity();
    if (mob instanceof Mob && mob.isSpawnCancelled())
```

Reply:
Given the statement `event.getEntity() instanceof Mob mob`, mob will be equivalent to `event.getEntity()` if the `instanceof` check passes, but with the cast already applied. If the `instanceof` fails, then `mob` will never be assigned, and the if statement will short-circuit.

Reply:
No, this syntax is shorthand version of your suggestion. They are functionally same.

Reply:
The problem is Optifine undoing this change.

Remove it.